### PR TITLE
Rationalise DCR's env checking scripts

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- moves DCR's `log`, `check-node` and `check-yarn` scripts to the project root
- moves `check-node-versions` script to DCR
- moves calling `check-deps` and `check-files` to DCR's makefile `lint-project` target
- removes unneeded `ensure`

There are also the following changes to how they work:

- updates `log` to output un-coloured messages outside of terminals
- converts `check-node` to a bash script
- updates `check-yarn` check for _any_ yarn, and recommends using `corepack enable` if none is available

## Why?

This is preparation work for updating and upgrading the way `yarn` and the workspace is used in the repo.

- `log`, `check-node` and `check-yarn` are not DCR-specific
- `check-node-versions` _is_ DCR-specific
- `check-node` currently runs in node, writing it as a bash script means it has no dependencies (e.g. node no longer comes with macOS).

### CODE REVIEW

File: dotcom-rendering/scripts/check-node-versions.mjs [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
 import { fileURLToPath } from 'node:url';
-import { log, warn } from '../dotcom-rendering/scripts/env/log.js';
+import { log, warn } from '../../scripts/log.js';
 
 const __dirname = dirname(fileURLToPath(import.meta.url));
 
 process.chdir(resolve(__dirname, '..'));
 
-const nvmrc = (await readFile('.nvmrc', 'utf-8'))
+const nvmrc = (await readFile('../.nvmrc', 'utf-8'))
```
Comment (mxdvl) [inline dotcom-rendering/scripts/check-node-versions.mjs:12]: I’m not sure this should live inside `dotcom-rendering` as it’s looking for files in both workspaces and in the root.

File: dotcom-rendering/scripts/env/ensure.js
Comment (mxdvl) [inline dotcom-rendering/scripts/env/ensure.js:1]: I believe this file no longer works, so it’s prime for removal.

File: scripts/env/check-yarn [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const { execSync } = require('child_process');
+const { stdout } = require('process');
+const { log, warn } = require('../log');
+
+const reset = '\x1b[0m';
+const blue = '\u001b[34m';
+
+try {
+	const yarnVersion = execSync('yarn -v', { encoding: 'utf-8' });
+	log(`Using yarn ${reset}${blue}${yarnVersion.trim()}`);
```
Comment (mxdvl) [inline scripts/env/check-yarn:22]: why not use `colourise` here?

File: scripts/env/check-yarn [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * use).
+ */
+
+// no external deps can be used here, because this runs before deps are installed
+const { execSync } = require('child_process');
+const { stdout } = require('process');
+const { log, warn } = require('../log');
+
+const reset = '\x1b[0m';
+const blue = '\u001b[34m';
```
Comment (mxdvl) [inline scripts/env/check-yarn:18]: Should we add a link to where we can get these magic strings, such as the [Wikipedia page for ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code#3-bit_and_4-bit)?

File: dotcom-rendering/scripts/check-node-versions.mjs (hunk 1 above)
Comment (sndrs) [reply dotcom-rendering/scripts/check-node-versions.mjs:12]: it's a DCR concern though right? it's looking in the root for the root version, and making sure that DCR-related files (which AR files are, really, i think?) are in sync with the repo more generally. 

should it be the responsibility of the root to ensure that apps in the workspace stay in sync with the root?

**Review by mxdvl — APPROVED** [review]
I like it!

It makes sense for `check-node` not to be written in JS.

File: dotcom-rendering/scripts/check-node-versions.mjs (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/scripts/check-node-versions.mjs:12]: In my mind the root is a good place for scripts that span all workspaces, but I agree that it should be a workspace concern to stay in sync with the root.

I’ll defer to anyone in @guardian/dotcom-platform with a stronger or better articulated view.

Comment (sndrs) [reply dotcom-rendering/scripts/check-node-versions.mjs:12]: Merged, but if anyone prefers to move it, please be my guest

**Pull request merged**